File: src/main/jvm/net/danielkvasnicka/neo4jtest/WebBean.scala
package net.danielkvasnicka.neo4jtest

import javax.inject.Inject
import javax.faces.bean.ManagedBean
import org.neo4j.graphdb.Node
import collection.JavaConversions._

/**
 * Created with IntelliJ IDEA.
 * User: daniel
 * Date: 22.06.13
 * Time: 14:14
 */
@ManagedBean
class WebBean {

  @Inject
  var dao: Dao = _

  def getNodes(): java.util.List[Node] = {
    dao.getNodes()
  }

  def addNode(): Unit = {
    dao.addNode()
  }
}

File: src/main/java/net/danielkvasnicka/neo4jtest/WebBean.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package net.danielkvasnicka.neo4jtest;

import java.util.Collection;
import javax.faces.bean.ManagedBean;
import javax.inject.Inject;
import org.neo4j.graphdb.Node;

/**
 *
 * @author daniel
 */
@ManagedBean
public class WebBean {
    
    @Inject
    private Dao dao;
    
    public Collection<Node> getNodes() {
        return this.dao.getNodes();
    }
    
    public void addNode() {
        this.dao.addNode();
    }

    public void setDao(Dao dao) {
        this.dao = dao;
    }
    
    
}
